feat(datasource/custom): allow to fetch from YAML data endpoint (#25695)
renovatebot/renovate: lib/modules/datasource/custom/readme.md

After the conversion, any `jsonata` rules defined in the `transformTemplates` section will be applied as usual to further process the JSON data.

> ### Yaml

Reviewer: This should be at h4 instead of h3 to be on the same as JSON and plain

> If `yaml` is used, response is processed by the `ts-yaml` library and then is converted into JSON.

Reviewer 2: ```suggestion
If `yaml` is used, response is parsed and converted into JSON for further processing.
```
